Source/WebCore/platform/audio/gstreamer/AudioEncoderGStreamer.cpp

```diff
@@ -139,34 +139,28 @@ GStreamerAudioEncoder::~GStreamerAudioEncoder()
     close();
 }
 
-void GStreamerAudioEncoder::encode(RawFrame&& frame, EncodeCallback&& callback)
+Ref<AudioEncoder::EncodePromise> GStreamerAudioEncoder::encode(RawFrame&& frame)
 {
-    gstEncoderWorkQueue().dispatch([frame = WTFMove(frame), encoder = m_internalEncoder, callback = WTFMove(callback)]() mutable {
+    return invokeAsync(gstEncoderWorkQueue(), [frame = WTFMove(frame), encoder = m_internalEncoder]() mutable {
         auto result = encoder->encode(WTFMove(frame));
         if (encoder->isClosed())
-            return;
-
-        String resultString;
-        if (result)
-            encoder->harness()->processOutputSamples();
```

Comment:
this part is no longer required? seems incorrect.

Reply:
Right, I need to add it back.